### PR TITLE
TPC-DS/TPCH-H stable results

### PR DESCRIPTION
**Pull request by ppalucha**
TPC-H and TPC-DS queries we use do not return stable results for larger scale factors. Main reason is there are multiple rows with the same values for columns used in ORDER BY clause. Therefore, I'm adding additional sorting column to assure stable results.

For query tpc-ds 39, source of instability is rounding when doing standard deviation / average calculations. PR limits precision of result by casting it to decimal type.

I'm also fixing a search/replace bug in tpc-ds query 77.

Fixing stability is required to use benchto results verification functionality.

### CODE REVIEW

**Comment by findepi** [comment]
Is it a deviation from TPC-H and TPC-DS definitions? is it a permissible one?

cc @sopel39 @kbajda

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql [hunk 1]
```diff
@@ -29,18 +29,18 @@ WITH
...
    WHERE ((CASE "mean" WHEN 0 THEN 0 ELSE ("stdev" / "mean") END) > 1)
-) 
+)
 SELECT
   "inv1"."w_warehouse_sk"
 , "inv1"."i_item_sk"
 , "inv1"."d_moy"
 , "inv1"."mean"
-, "inv1"."cov"
+, CAST("inv1"."cov" AS DECIMAL(30, 10))
```
Comment (sopel39) [inline testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql:38]: this requires comment

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql [hunk 2]
```diff
@@ -29,18 +29,18 @@ WITH
...
 , "inv2"."w_warehouse_sk"
 , "inv2"."i_item_sk"
 , "inv2"."d_moy"
 , "inv2"."mean"
-, "inv2"."cov"
+, CAST("inv2"."cov" AS DECIMAL(30, 10))
```
Comment (sopel39) [inline testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql:43]: ditto

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q65.sql [hunk 3]
```diff
@@ -43,5 +43,5 @@ WHERE ("sb"."ss_store_sk" = "sc"."ss_store_sk")
    AND ("sc"."revenue" <= (DECIMAL '0.1' * "sb"."ave"))
    AND ("s_store_sk" = "sc"."ss_store_sk")
    AND ("i_item_sk" = "sc"."ss_item_sk")
-ORDER BY "s_store_name" ASC, "i_item_desc" ASC
+ORDER BY "s_store_name" ASC, "i_item_desc" ASC, "i_brand" ASC, "sc"."revenue" ASC
```
Comment (sopel39) [inline testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q65.sql:46]: this requires comment

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql [hunk 4]
```diff
@@ -38,7 +38,7 @@ WITH
    WHERE ("cs_sold_date_sk" = "d_date_sk")
       AND ("d_date" BETWEEN CAST('2000-08-23' AS DATE) AND (CAST('2000-08-23' AS DATE) + INTERVAL  '30' DAY))
    GROUP BY "cs_call_center_sk"
-)
```
Comment (sopel39) [inline testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql:41]: nit: make formatting fixes a separate commit

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql [hunk 5]
```diff
@@ -88,7 +88,7 @@ SELECT
 FROM
   (
    SELECT
-     '${database}.${schema}.store channel' "channel"
```
Comment (sopel39) [inline testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql:91]: nit: separate commit

**Comment by sopel39** [comment]
> Is it a deviation from TPC-H and TPC-DS definitions? is it a permissible one?

I think it's fine as it shouldn't affect (too much) benchmarks results but allow for query result validation.

However, we should put a comment with warning in each of such changes to easily identify them if needed

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql (hunk 1 above)
Comment (ppalucha) [reply testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql:38]: Comments are included in commit message - do you want them in code?

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql (hunk 5 above)
Comment (ppalucha) [reply testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql:91]: This is a separate commit (apart from unintentional whitespace changes - I will revert them).

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql (hunk 1 above)
Comment (sopel39) [reply testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q39.sql:38]: > Comments are included in commit message - do you want them in code?

Yes.

**Comment by ppalucha** [comment]
> > Is it a deviation from TPC-H and TPC-DS definitions? is it a permissible one?
> 
> I think it's fine as it shouldn't affect (too much) benchmarks results but allow for query result validation.
> 
> However, we should put a comment with warning in each of such changes to easily identify them if needed

Also, our queries are described as "closely following TPC specifications"; we already have multiple deviations from specifications - e.g. we always use constant queries, while specification requires to use random elements for some of the queries.

File: testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql (hunk 4 above)
Comment (ppalucha) [reply testing/trino-benchto-benchmarks/src/main/resources/sql/presto/tpcds/q77.sql]: Removed formatting changes.

**Review by sopel39 — APPROVED** [review]
